Feature/comms

afrojaA opened the pull request:
Added some basic code for: 

>comms() constructor
> calibrate()
>sendCommand(char type, int data)
> receivedCommand()

afrojaA committed:
Fixed merged conflicts

afrojaA committed:
Starter code for the Comms Subsystem

Basic code for:
> comms() constructor
> calibrate()
>sendCommand(char type, int data)
>receivedCommand()

chowii commented:
looks good. nice first one

TrideepLD reviewed (approved):
Looks good to me 👍

natlec commented on MEGA/Main.ino:
```diff
@@ -47,15 +47,33 @@ State state();
       int baudRate = 9600;
 
     public:
+      char command;
```
Might be best to make this a private variable (under `private:`) considering it'll only be accessed in the Comms class

natlec commented on MEGA/Main.ino:
```diff
@@ -47,15 +47,33 @@ State state();
     public:
+      char command;
       Comms() {
         Serial.begin(baudRate);
```
We'll call `Serial.begin()` in the setup function once, no need to call it in the constructor for each subsystem class.

natlec commented on MEGA/Main.ino:
```diff
@@ -47,15 +47,33 @@ State state();
         Send test command over comms and check that response is correct
       */
-      void calibrate() {
+      bool calibrate() {
```
Will need to communicate our calibration functionality with @natleco/t1_c5

natlec reviewed (commented):
Reviewed, looks good pending address of comments

afrojaA committed:
Addressed the reviews from previous PR

-For Comms() constructor, it will only include Serial.begin.
- The baud rate has been defined rather than including it as a variable. 
-The char command has been kept under private as it needs to be accessed by the comms only
